httpx/client.py

```diff
@@ -257,6 +258,14 @@ def wrap(
     def _get_auth_middleware(
         self, request: AsyncRequest, trust_env: bool, auth: AuthTypes = None
     ) -> typing.Optional[BaseMiddleware]:
+        _NETRC = None
+
+        def _get_netrc_login(host: str) -> typing.Optional[typing.Tuple[str, str, str]]:
```

Comment:
Could we keep `get_netrc_login(host)` in `utils` and use `get_netrc()` within that function? That way the implementation of `get_netrc_login()` doesn't have to leak into the Client like this.